include/SFML/Window/Keyboard.hpp

```diff
@@ -153,6 +158,158 @@ class SFML_WINDOW_API Keyboard
+    enum Scancode
+    {
+        ScanUnknown = -1,      ///< Represents any scancode not present in this enum
+        ScanA = 0,             ///< Keyboard a and A key
+        ScanB,                 ///< Keyboard b and B key
+        ScanC,                 ///< Keyboard c and C key
+        ScanD,                 ///< Keyboard d and D key
+        ScanE,                 ///< Keyboard e and E key
+        ScanF,                 ///< Keyboard f and F key
+        ScanG,                 ///< Keyboard g and G key
+        ScanH,                 ///< Keyboard h and H key
+        ScanI,                 ///< Keyboard i and I key
+        ScanJ,                 ///< Keyboard j and J key
+        ScanK,                 ///< Keyboard k and K key
+        ScanL,                 ///< Keyboard l and L key
+        ScanM,                 ///< Keyboard m and M key
+        ScanN,                 ///< Keyboard n and N key
+        ScanO,                 ///< Keyboard o and O key
+        ScanP,                 ///< Keyboard p and P key
+        ScanQ,                 ///< Keyboard q and Q key
+        ScanR,                 ///< Keyboard r and R key
+        ScanS,                 ///< Keyboard s and S key
+        ScanT,                 ///< Keyboard t and T key
+        ScanU,                 ///< Keyboard u and U key
+        ScanV,                 ///< Keyboard v and V key
+        ScanW,                 ///< Keyboard w and W key
+        ScanX,                 ///< Keyboard x and X key
+        ScanY,                 ///< Keyboard y and Y key
+        ScanZ,                 ///< Keyboard z and Z key
+        ScanNum1,              ///< Keyboard 1 and ! key
+        ScanNum2,              ///< Keyboard 2 and @ key
+        ScanNum3,              ///< Keyboard 3 and # key
+        ScanNum4,              ///< Keyboard 4 and $ key
+        ScanNum5,              ///< Keyboard 5 and % key
+        ScanNum6,              ///< Keyboard 6 and ^ key
+        ScanNum7,              ///< Keyboard 7 and & key
+        ScanNum8,              ///< Keyboard 8 and * key
+        ScanNum9,              ///< Keyboard 9 and ) key
+        ScanNum0,              ///< Keyboard 0 and ) key
+        ScanEnter,             ///< Keyboard Return (ENTER) key
+        ScanEscape,            ///< Keyboard Escape key
+        ScanBackspace,         ///< Keyboard Backspace key
+        // TODO above it's BackSpace, but is it correct? What do we use here?
+        ScanTab,               ///< Keyboard Tab key
+        ScanSpace,             ///< Keyboard Space key
+        ScanHyphen,            ///< Keyboard - and _ key
+        ScanEquals,            ///< Keyboard = and +
+        ScanLBracket,          ///< Keyboard [ and { key
+        ScanRBracket,          ///< Keyboard ] and } key
+        ScanBackslash,         ///< Keyboard \ and | key
+        // TODO capitalisation
+        ScanDash,              ///< Keyboard Non-US # and ~
```

Comment:
After some reasearch, it looks like a proper name of that key is "[hyphen-minus](https://en.wikipedia.org/wiki/Hyphen-minus)". But it's certainly not a dash, because typing a dash (even en/em dashes) is not possible with a regular keyboard.

Scancode may be called ScanHyphen or ScanMinus (though we have ScanMinus for numpad...), so we need to discuss this as well. sf::Keyboard::Key::Dash also exists and probably should be renamed.

Reply:
`ScanHyphen` makes more sense to me. For scancode it's not an issue to make the difference between hyphen and minus (through numpad), but for `Keyboard` codes it's harder (impossible?)... Maybe we should deprecate both of `Dash` and `Subtract` and always map it to a new `HyphenMinus`? Any opinion @LaurentGomila?

Reply:
I don't know. Maybe we should just have a look at what others do.

Reply:
GLFW uses "minus": GLFW_KEY_MINUS
SDL is the same: SDL_SCANCODE_MINUS
Any other examples?

Reply:
@eliasdaler @LaurentGomila What do we pick now and for which key?

Reply:
Maybe due to the merging of the other branches, this has become a bit mixed up, or maybe it  was always in confusion. For scancode 0x0C we'll be using `ScanHyphen`.

The discussion is about key 42 which is not on the US keyboard layout, but since Windows is actually return the scancode 0x2B (ScanBackslash), I'm not sure if we need to find a different name for it, basically trading place with key 29, as that one is only available on US keyboards. As such, I believe the discussion for `ScanDash` becomes obsolete, as it's not going to be used anyways, at least on Windows, we'll have to confirm this on other OS.

image:
![image](https://user-images.githubusercontent.com/920861/84440249-f0b92400-ac39-11ea-9843-6be970387f94.png)